src/provision.toit

```diff
@@ -8,24 +8,36 @@ import monitor
 import net.wifi
 import ble
 import protobuf
+import .srp
+import crypto.aes show *
 
 /** This UUID is used in PC and Phone APP by default. */
 SERVICE_UUID ::= #[0x02, 0x1a, 0x90, 0x04, 0x03, 0x82, 0x4a, 0xea,
                    0xbf, 0xf4, 0x6b, 0x3f, 0x1c, 0x5a, 0xdf, 0xb4]
 
-/** This security mode doesn't encrypt/decrypt. */
+/** This security mode 0 doesn't encrypt/decrypt. */
 SECURITY0 := Security0_
 
+/** This security mode 2 use SRP6a + AES-GCM*/
+SECURITY2 salt verifier:
+  return Security2_ salt verifier
+
 class BLECharacteristic_:
   characteristic/ble.LocalCharacteristic
+  encrypted_/bool
+  desc/string
+  recv_task/Task? := null
+  result/ByteArray := #[]
+  sem := monitor.Semaphore --count=1 --limit=1
```

Comment:
Semaphore seems like the wrong synchronization tool here.

I think a mutex would work better.
```suggestion
  mutex_ := monitor.Mutex
```